homeassistant/components/flipr/entity.py

```diff
@@ -17,17 +15,21 @@ class FliprEntity(CoordinatorEntity):
     _attr_has_entity_name = True
 
     def __init__(
-        self, coordinator: DataUpdateCoordinator, description: EntityDescription
+        self,
+        coordinator: FliprDataUpdateCoordinator,
+        description: EntityDescription,
+        is_flipr_hub: bool = False,
     ) -> None:
         """Initialize Flipr sensor."""
         super().__init__(coordinator)
+        self.device_id = coordinator.device_id
         self.entity_description = description
-        if coordinator.config_entry:
-            flipr_id = coordinator.config_entry.data[CONF_FLIPR_ID]
-            self._attr_unique_id = f"{flipr_id}-{description.key}"
+        self._attr_unique_id = f"{self.device_id}-{description.key}"
 
-            self._attr_device_info = DeviceInfo(
-                identifiers={(DOMAIN, flipr_id)},
-                manufacturer=MANUFACTURER,
-                name=f"Flipr {flipr_id}",
-            )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device_id)},
+            manufacturer=MANUFACTURER,
+            name=f"Flipr hub {self.device_id}"
+            if is_flipr_hub
+            else f"Flipr {self.device_id}",
```

Comment:
Does the hub expose entities?

Reply:
Yes : Hub has a switch and a select platform.

Reply:
Do we know what devices are connected with what hub?

Reply:
No. The hub is a big switch.
And behind, the user can connect the pump or a light. The hub device does not know that.

Reply:
so we can't link the devices to the hub they are connected to?